src/pages/register-a-beacon/check-beacon-details.tsx

```diff
@@ -16,82 +16,44 @@ import { Details } from "../../components/Details";
 import { IfYouNeedHelp } from "../../components/Mca";
 import { GetServerSideProps, GetServerSidePropsContext } from "next";
 import { BeaconCacheEntry } from "../../lib/formCache";
+import { updateFormCache, withCookieRedirect } from "../../lib/middleware";
 import {
-  getCache,
-  updateFormCache,
-  withCookieRedirect,
-} from "../../lib/middleware";
-import { ErrorSummary } from "../../components/ErrorSummary";
-import { FieldValidator } from "../../lib/fieldValidator";
+  FormErrorSummary,
+  FieldErrorList,
+} from "../../components/ErrorSummary";
+
+import { FormValidator } from "../../lib/formValidator";
+import { Beacon } from "../../lib/types";
+import { ensureFormDataHasKeys } from "../../lib/utils";
 
 interface CheckBeaconDetailsProps {
-  manufacturer: string;
-  model: string;
-  hexId: string;
+  formData: Partial<Beacon>;
```

Comment:
Can we use the type `BeaconCacheEntry` here which is essentially the convenience type we defined in `types.ts`

Reply:
Done :-)